sdks/java/io/tika/src/main/java/org/apache/beam/sdk/io/tika/TikaIO.java

```diff
@@ -0,0 +1,307 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.tika;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import com.google.auto.value.AutoValue;
+
+import javax.annotation.Nullable;
+
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.io.Read.Bounded;
+import org.apache.beam.sdk.options.ValueProvider;
+import org.apache.beam.sdk.options.ValueProvider.StaticValueProvider;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.display.DisplayData;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.tika.metadata.Metadata;
+
+
+/**
+ * {@link PTransform} for parsing arbitrary files using Apache Tika.
+ * Files in many well known text, binary or scientific formats can be processed.
+ *
+ * <p>To read a {@link PCollection} from one or more files
+ * use {@link TikaIO.Read#from(String)}
+ * to specify the path of the file(s) to be read.
+ *
+ * <p>{@link TikaIO.Read} returns a bounded {@link PCollection} of {@link String Strings},
+ * each corresponding to a sequence of characters reported by Apache Tika SAX Parser.
```

Comment:
Does this correspond to any particular method in the Tika API? It seems like the reader returns key-value pairs reported as key"="value which begs for returning KVs instead of Strings. Moreover, if you apply this to filepatterns, it returns them as an unordered soup - metadata items for all files at the same time in no particular order. What is the use case for such an API?

Reply:
Tika reports the data via the SAX ContentHandler callbacks, as a sequence of char fragments. This is the main use case, getting the content and adapting it to the Beam pipeline read flow. Tika can also populate the metadata (not the main content), after it has reported the content, and thus, optionally, TikaIO can report it too. The main content is String; you are right, the metadata itself could've been reported differently, but I guess it would make things quite complex given that in most cases I believe the users would be after a (main) String content only, and they can easily split the optional follow-up metadata if they need it.